### PR TITLE
Extend Profile Editor

### PR DESCRIPTION
**Pull request by jasoncalabrese**
Building on @MilosKozak's #782 this will add some basic tests, link from the settings page, and other minor changes

### CODE REVIEW

**Comment by jasoncalabrese** [comment]
next is to replace the `XMLHttpRequest` based post of the mockable `$.ajax` post, then we can verify post and by tinkering with the dom to simulate user edits.

File: static/profile/js/profile.js [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+    }
+    
+    function shouldAddTime(i,time,array) {
+      if (i===0 && time>0) { return false; }
+      if (i>0 && isNaN(toMinutesFromMidnight(c_profile[array][i-1].time))) { return false; }
+      if (i>0 && toMinutesFromMidnight(c_profile[array][i-1].time) >= time*30) { return false; } 
+      return true;
+    }
+ 
+    function addSingleLine(e,i) {
```
Comment (jasoncalabrese) [inline static/profile/js/profile.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/859032400/issues/source?bid=2452379&fileBranchId=2815005#l228)

File: static/profile/js/profile.js [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+    $('.delsingle').click(function delsingle_click() {
+      var array = $(this).attr('array');
+      var pos = $(this).attr('pos');
+      GUIToObject(); 
+      c_profile[array].splice(pos,1);
+      c_profile[array][0].time = '00:00';
+      return fillTimeRanges();
+    });
+
+    function addBGLine(i) {
```
Comment (jasoncalabrese) [inline static/profile/js/profile.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/859032400/issues/source?bid=2452379&fileBranchId=2815005#l291)

**Comment by jasoncalabrese** [comment]
### Added link from settings if any related plugins are enabled

screen shot 2015-08-15 at 2 03 18 pm

**Comment by francescaneri** [comment]
@jasoncalabrese  you can add the plug-in battery?

**Pull request merged**